extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/BeanParamTest.java

```diff
@@ -0,0 +1,104 @@
+package io.quarkus.resteasy.reactive.server.test.beanparam;
+
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class BeanParamTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> {
+                return ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(MyBeanParamWithFieldsAndProperties.class, Top.class);
+            });
+
+    @Test
+    void shouldDeployWithoutIssues() {
+        // we only need to check that it deploys
```

Comment:
@FroMage you could also check that the bean has all the expected types, see [`ResourceBeanTypeTest`](https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanTypes/ResourceBeanTypeTest.java#L39-L45) as an example.
But I won't insist :)